server/entity/experience.go

```diff
@@ -0,0 +1,102 @@
+package entity
+
+var maxLevel = 21474836477
+
+type ExperienceManager struct {
+	level    int
+	progress float64
+	totalXP  int
+}
+
+func NewExperienceManager() *ExperienceManager {
+	return &ExperienceManager{
+		0,
+		0,
+		0,
+	}
+}
+func (e *ExperienceManager) Level() int {
+	return e.level
+}
+func (e *ExperienceManager) Progress() float64 {
+	return e.progress
+}
+func (e *ExperienceManager) MaxLevel() int {
+	return maxLevel
+}
+func (e *ExperienceManager) TotalXP() int {
+	return e.totalXP
+}
+
+// SetTotalXP internal
+func (e *ExperienceManager) SetTotalXP(amount int) {
+	e.totalXP = amount
```

Comment:
Do you not need to recalculate the progress?

Reply:
i have remove useless function